### PR TITLE
feat: non-JSON bindings between Rust and Swift

### PR DESCRIPTION
**Pull request by michaelx11**
## Overview

This marks the start of v0.2.0 series, which utilizes a more efficient non-JSON binding scheme between Rust and Swift. This repo can be consumed as a Swift package or Cocoapod by [xmtp-ios](https://github.com/xmtp/xmtp-ios) and replaces:

- GRPC dependencies (not compatible with Cocoapods)
- Selected cryptographic operations (Swift packages utilized are not compatible with Cocoapods)

This allows `xmtp-ios` to itself be packaged as a Cocoapod and consumed within a React Native SDK or application.

## Project Status

This is still a prerelease version. Non-XMTP engineers should await an official `XMTP` Cocoapod release from the `xmtp-ios` repository.

## Breaking Changes

- No more ApiService.swift
- No more CoreCrypto.swift

Examples:
```
let client = try await XMTPRust.create_client(GRPCApiClient.envToUrl(env: options.api.env), options.api.env != .local)

// Subcode in ApiClient
...
let envelopesVec = RustVec<XMTPRust.Envelope>()

envelopes.forEach( { envelope in
    et rustEnvelope = XMTPRust.create_envelope(envelope.contentTopic.intoRustString(), envelope.timestampNs, dataToRustVec(data: envelope.message))
    envelopesVec.push(value: rustEnvelope)
})
let response = try await self.rustClient.publish(self.authToken.intoRustString(), envelopesVec)
...
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,51 @@
...
-│    ---> bindingx/xmtp_rust_swift   │  thin wrapper of XMTPRust   │- Consumes xmtp/proto on branch  │
-│                                    │    ┌───────────────────┐    │   without gRPC Swift files      │
-│ ┌──────────┐     ┌───────────────┐ │ ┌─►│-Package.swift     │    │- Key: no more GRPC dep!         │
-│ │xmtp-proto├────►│xmtp-networking│ │ │  │-XMTPRust.xcframe..│    │    ┌──────────────────┐         │
-│ └──────────┘     └─────┬─────────┘ │ │  │-Sources/...       │    │    │                  │         │
-│                        │           │ │  │ -ApiService.swift │    │    │ Package.swift    │         │
-│        ┌───────────────▼────┐      │ │  └─────────┬─────────┘    │ ┌──►                  │         │
-│        │libxmtp/bindings/   │      │ │            │  Swift Pkg   │ │  │ - ApiClient.swift│         │
-│        │  xmtp_rust_swift   │      │ │            └──────────────┼─┘  │  - Uses XMTPRust │         │
+│  xmtp/libxmtp: Shared Rust Code    │  xmtp/xmtp-rust-swift       │ xmtp/xmtp-ios - iOS SDK         │
```
Comment (neekolas) [inline README.md:29]: Curious what you've been using to create these ASCII diagrams

File: Sources/XMTPRust/SwiftBridgeCore.swift [hunk 2]
```diff
@@ -1,4 +1,6 @@
+import Foundation
```
Comment (neekolas) [inline Sources/XMTPRust/SwiftBridgeCore.swift:1]: Looks like a dupe import here

**Review by neekolas — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (michaelx11) [reply README.md:29]: I use https://asciiflow.com/#/ then manually shift things around in Vim

File: Sources/XMTPRust/SwiftBridgeCore.swift (hunk 2 above)
Comment (michaelx11) [reply Sources/XMTPRust/SwiftBridgeCore.swift:1]: Good eyes, I need to go revisit my import injection code in libxmtp

Comment (michaelx11) [reply Sources/XMTPRust/SwiftBridgeCore.swift:1]: https://github.com/xmtp/libxmtp/pull/71

**Pull request merged**